cdap-ui/app/features/hydrator/services/non-store-pipeline-error-factory.js

```diff
@@ -198,6 +198,89 @@ let allNodesConnected = (GLOBALS, nodes, connections, cb) => {
   });
 };
 
+let hasValidArtifact = (importConfig) => {
+  return importConfig.artifact && importConfig.artifact.name.length && importConfig.artifact.version.length && importConfig.artifact.scope.length;
+};
+let hasValidConfig = (importConfig) => {
+  return importConfig.config.source && importConfig.config.sinks.length && importConfig.config.connections.length;
+};
+let hasValidArtifactsForNodes = (importConfig) => {
+  let config = importConfig.config;
+  return [config.source].concat(config.sinks)
+    .concat( (config.transforms || []) )
+    .filter(node => !node.plugin.artifact)
```

Comment:
do we want to enforce this? Technically the backend does not enforce plugins artifacts. If none specified, it will use the latest